build.gradle

```diff
@@ -1,16 +1,57 @@
 plugins {
     id "com.dorongold.task-tree" version "2.1.1"
     id 'io.freefair.lombok' version '8.6' apply false
+    id "com.diffplug.spotless" version "7.0.0.BETA1"
 }
 
 task buildDockerImages() {
     dependsOn(':TrafficCapture:dockerSolution:buildDockerImages')
     dependsOn(':DocumentsFromSnapshotMigration:buildDockerImages')
 }
 
+spotless {
+    format 'misc', {
+        target '**/*.gradle', '.gitattributes', '.gitignore'
+        trimTrailingWhitespace()
+        indentWithSpaces()
+        endWithNewline()
+    }
+    yaml {
+        target '**/*.yml'
+        prettier()
+        trimTrailingWhitespace()
+        indentWithSpaces()
+        endWithNewline()
+    }
+    python {
+        target "**/*.py"
+        trimTrailingWhitespace()
+        indentWithSpaces()
+        endWithNewline()
+    }
+}
+
 subprojects {
     apply plugin: 'java'
     apply plugin: 'maven-publish'
+    apply plugin: "com.diffplug.spotless"
+
+    spotless {
+        java {
```

Comment:
There is a lot of flexibility with the java styling, we can update options here and in the `formatter.xml` to tweak any settings that are must haves.